tools/generate_schema_wrapper.py

```diff
@@ -861,59 +865,72 @@ def vegalite_main(skip_download: bool = False) -> None:
 
 
-def _create_encode_signature(
-    channel_infos: dict[str, ChannelInfo],
-) -> str:
+def generate_encoding_artifacts(
+    channel_infos: dict[str, ChannelInfo], fmt_method: str, fmt_typed_dict: str
+) -> Iterator[str]:
+    """
+    Generate `.encode()` related things.
+
+    This was previously `_create_encode_signature()`. Now also creates a `TypedDict` as
+    part of https://github.com/pola-rs/polars/pull/17995.
+
+    Notes
+    -----
+    - `Map`/`Dict` stands for the return types of `alt.(datum|value)`, and any encoding channel class.
+        - See discussions in https://github.com/vega/altair/pull/3208
+    - We could be more specific about what types are accepted in the `List`
+        - but this translates poorly to an IDE
+        - `info.supports_arrays`
+    """
     signature_args: list[str] = ["self", "*args: Any"]
-    docstring_parameters: list[str] = ["", "Parameters", "----------"]
+    signature_docstring_parameters: list[str] = ["", "Parameters", "----------"]
+    channel_type_aliases: list[str] = []
+    typed_dict_docstring_parameters: list[str] = ["", "Parameters", "----------"]
+    typed_dict_args: list[str] = []
     for channel, info in channel_infos.items():
-        field_class_name = info.field_class_name
-        assert (
-            field_class_name is not None
-        ), "All channels are expected to have a field class"
-        datum_and_value_class_names = []
-        if info.datum_class_name is not None:
-            datum_and_value_class_names.append(info.datum_class_name)
-
-        if info.value_class_name is not None:
-            datum_and_value_class_names.append(info.value_class_name)
-
-        # dict stands for the return types of alt.datum, alt.value as well as
-        # the dictionary representation of an encoding channel class. See
-        # discussions in https://github.com/vega/altair/pull/3208
-        # for more background.
-        union_types = ["str", field_class_name, "Map"]
-        docstring_union_types = ["str", rst_syntax_for_class(field_class_name), "Dict"]
+        it = info.all_names
+        it_rst_names = (rst_syntax_for_class(c) for c in info.all_names)
+
+        docstring_union_types = ["str", next(it_rst_names), "Dict"]
+        tp_inner: str = "Union[" + ", ".join(chain(("str", next(it), "Map"), it)) + "]"
         if info.supports_arrays:
-            # We could be more specific about what types are accepted in the list
-            # but then the signatures would get rather long and less useful
-            # to a user when it shows up in their IDE.
-            union_types.append("list")
             docstring_union_types.append("List")
+            tp_inner = f"OneOrSeq[{tp_inner}]"
 
-        union_types = union_types + datum_and_value_class_names
-        docstring_union_types = docstring_union_types + [
-            rst_syntax_for_class(c) for c in datum_and_value_class_names
-        ]
+        alias_name = f"Channel{channel[0].upper()}{channel[1:]}"
```

Comment:
How're you planning to get each `alias_name` into `altair.typing` without including it in `channels.__all__`?

Reply:
I'm now trying to rename them to `_ChannelX` and then import directly in `altair.typing` and expose it there via `__all__`. Same as for `EncodeKwds`. Don't think it's worth it right now to autogenerate `altair.typing` as there are very rarely new additions of channels but would of course be a nice-to-have. What do you think?

Reply:
I'll also add it to the API docs so that it's declared as public. I will finish and push it tomorrow.

Reply:
> I'm now trying to rename them to `_ChannelX` and then import directly in `altair.typing` and expose it there via `__all__`. Same as for `EncodeKwds`. Don't think it's worth it right now to autogenerate `altair.typing` as there are very rarely new additions of channels but would of course be a nice-to-have. What do you think?

You can keep the name the same btw. They won't be added to `channels.__all__` unless you write code to do that. 

Just thinking writing out 40ish renames in `typing` might be a lot

Reply:
👍 I've now also renamed `_EncodeKwds` to `EncodeKwds` as else it shows up like this in Sphinx (could probably also be fixed but now it's also consistent with the `Channel...` type aliases): 
<img width="528" alt="image" src="https://github.com/user-attachments/assets/f0fbb964-99f7-496e-b381-065e5a56e20e">

In the latest commit, I've also:
* Added channel type aliases to typing module
* Added 'Type Hints' section to API references in docs.
* Removed is_chart_type from 'API' docs section as it's now in typing. Rename '_EncodeKwds' to 'EncodeKwds'

Would you mind having another look?

Reply:
> Would you mind having another look?

@binste Wow excellent!

Yeah I will go through it holistically later today, but good work on those points 👍

Reply:
@binste I do have some things I wanna discuss before making any changes other than https://github.com/vega/altair/pull/3515#commits-pushed-b3fbe9c

But overall I think we're very close to finished on this 😄